crypto/src/hash/pedersen/mod.rs

```diff
@@ -0,0 +1,136 @@
+use super::traits::IsCryptoHash;
+use lambdaworks_math::{
+    cyclic_group::IsGroup,
+    elliptic_curve::{
+        self,
+        short_weierstrass::{
+            curves::bls12_381::{curve::BLS12381Curve, field_extension::BLS12381PrimeField},
+            element::ProjectivePoint,
+        },
+        traits::IsEllipticCurve,
+    },
+    field::{self, element::FieldElement, traits::IsField},
+    traits::ByteConversion,
+    unsigned_integer::element::U384,
+};
+
+type FE = field::element::FieldElement<BLS12381PrimeField>;
+type Point = ProjectivePoint<BLS12381Curve>;
+
+const WINDOW_SIZE: usize = 4;
+const NUM_WINDOWS: usize = 96;
+const INPUT_SIZE_IN_BITS: usize = WINDOW_SIZE * NUM_WINDOWS;
+const HALF_INPUT_SIZE_BITS: usize = INPUT_SIZE_IN_BITS / 2;
+
+pub fn random_point(rng: &mut rand::rngs::ThreadRng) -> Point {
+    BLS12381Curve::generator().operate_with_self(rand::Rng::gen::<u128>(rng))
+}
+
+// TODO: this function should be replaced with the trait method once it is implemented.
+pub fn random_field_element<F>(rng: &mut rand::rngs::ThreadRng) -> FieldElement<F>
+where
+    F: field::traits::IsField,
+{
+    FieldElement::<F>::from(rand::Rng::gen::<u64>(rng))
+}
+
+pub struct Pedersen<E>
+where
+    E: elliptic_curve::short_weierstrass::traits::IsShortWeierstrass,
```

Comment:
Is there any specific reason why the curve needs to be `IsShortWeierstrass` or would `IsEllipticCurve` be enough? Not sure if there is any specific requirement I'm missing.

Reply:
Nope, I'll change it